Lib/codeop.py

```diff
@@ -86,24 +87,12 @@ def _maybe_compile(compiler, source, filename, symbol):
     with warnings.catch_warnings():
         warnings.simplefilter("error")
 
-        code1 = err1 = err2 = None
         try:
-            code1 = compiler(source + "\n", filename, symbol)
+            code1 = compiler(source + "\n", filename, symbol, flags=ast.PyCF_ALLOW_INCOMPLETE_INPUT)
         except SyntaxError as e:
-            err1 = e
-
-        try:
-            code2 = compiler(source + "\n\n", filename, symbol)
-        except SyntaxError as e:
-            err2 = e
-
-    try:
-        if not code1 and _is_syntax_error(err1, err2):
-            raise err1
-        else:
-            return None
-    finally:
-        err1 = err2 = None
+            if "incomplete input" in str(e):
+                return None
+            raise
```

Comment:
Yeah, the idea of this draft PR is to remove all hacks, even as fallbacks. Otherwise we are going to shoot ourselves in the foot again maintaining features that are working almost by chance and make almost impossible to improve error messages

Reply:
No hack is what I want.